src/image_function.cpp

```diff
@@ -858,23 +858,27 @@ namespace Image_Function
     void Histogram( const Image & image, uint32_t x, uint32_t y, uint32_t width, uint32_t height, std::vector < uint32_t > & histogram )
     {
         ParameterValidation( image, x, y, width, height );
-        VerifyGrayScaleImage( image );
         OptimiseRoi( width, height, image );
 
-        histogram.resize( 256u );
+        const uint8_t colorCount = image.colorCount();
+
+        histogram.resize( 256u * colorCount );
         std::fill( histogram.begin(), histogram.end(), 0u );
 
         const uint32_t rowSize = image.rowSize();
 
-        const uint8_t * imageY = image.data() + y * rowSize + x;
+        const uint8_t * imageY = image.data() + y * rowSize + x * colorCount;
         const uint8_t * imageYEnd = imageY + height * rowSize;
 
         for( ; imageY != imageYEnd; imageY += rowSize ) {
             const uint8_t * imageX    = imageY;
             const uint8_t * imageXEnd = imageX + width;
 
-            for( ; imageX != imageXEnd; ++imageX )
-                ++histogram[*imageX];
+            for( ; imageX != imageXEnd; imageX += colorCount ) {
+                uint8_t colorChannel = 0;
+                for( ; colorChannel < colorCount; colorChannel++ )
+                    ++histogram[*(imageX + colorChannel)];
```

Comment:
Your implementation will always fill 256 values of the array.

Reply:
Let's do this way:
```cpp
if ( colorCount  == 1 ) {
    // put original double loop here
} else {
    loop1
        loop2
            ++histogram[*(imageX + colorChannel) * colorCount + colorChannel];
}

Reply:
Oh yes you are right there is something wrong with `++histogram[...]`.
For the loops, sure thing I can fix that but can you please help me understand the benefit of splitting the cases? I thought it boils down to the same thing in case `colorCount` is 1.

Reply:
The benefit is performance only. Most of images with which people work are gray-scale aka 1 color channel image so it's pretty important to have highest performance.